### DISCUSSION TITLE
Picker Option Exclusivity?

### DISCUSSION THREAD

**Discussion by CosmasSC128** [Budibase/budibase#4846] [Help] [answered]
I am trying to set up multiple pickers on a form, where if an option is selected on one, that option is not available for the other pickers on the form. Is this possible?

**Comment by melohagan** [comment, marked as answer]
Hi @CosmasSC128 

This is possible! You can use multiple data providers and filters as follows:

1. Add a Parent Data Provider to your form. In my case I linked it to a table with a list of fruit and veg.

2. Add the Parent Option Picker which will control the filters on the children. Set the options source to point to the parent data provider.

![Screenshot 2022-03-24 at 11 28 40](https://user-images.githubusercontent.com/101575380/159906961-a386b58f-93dc-4b4d-9fd8-2b535136dc7b.png)

It's important in this step to enter **ParentPicker** into the field input. This will be used to reference the selected value of the parent picker by the children.

3. Add a Child Data Provider to your form. Link this to the parent. 

![Screenshot 2022-03-24 at 11 30 01](https://user-images.githubusercontent.com/101575380/159907180-e5977b06-4a16-49de-b343-54ab9fb5c80b.png)

4. Click define filters. Select your label column, not equals to the binding: `{{ Form.Fields.ParentPicker }}`

![Screenshot 2022-03-24 at 11 31 43](https://user-images.githubusercontent.com/101575380/159907423-f68fdf3b-e5bf-4a90-820c-44cd4d3ae0d8.png)

5. Finally, add your Child Option Picker(s) and set the options source to the child data provider:

![Screenshot 2022-03-24 at 11 42 43](https://user-images.githubusercontent.com/101575380/159909120-665d1135-d6d5-4cce-9321-d23c51b8e9ec.png)

<table><tr><td><img src="https://user-images.githubusercontent.com/101575380/159909378-17690543-f508-4f08-846b-795c0cec6c62.png" /></td>
<td> <img src="https://user-images.githubusercontent.com/101575380/159909439-1e6b0a59-da2a-4501-b2ca-afc625b0ca80.png" /></td></tr></table>



----

One thing to note, is that your app could get into an invalid state:

1. Select option 'Apple' on parent
2. Select option 'Banana' on child
3. Select option 'Banana' on parent.

One work around is to use Form Validation. For your Child Picker, configure validation as follows:

![Screenshot 2022-03-24 at 11 56 45](https://user-images.githubusercontent.com/101575380/159911280-d12a352d-0d23-46c6-afbc-c52b377aa14f.png)

Then you can add a 'Submit' button to your form. Define actions -> Validate Form. 

More on forms here: https://docs.budibase.com/docs/forms

Hopefully this is of some use!

**Comment by CosmasSC128** [comment]
Thank you for the response.
